src/System.Globalization.Extensions/tests/IdnMapping/Data/Unicode_6_0/Unicode_6_0_IdnaTest.cs

```diff
@@ -69,7 +68,10 @@ private static string EscapedToLiteralString(string escaped, int lineNumber)
                 if (i + 1 < escaped.Length && escaped[i] == '\\' && escaped[i + 1] == 'u')
                 {
                     // Verify that the escaped sequence is not malformed
-                    Assert.True(i + 5 < escaped.Length, "There was a problem converting to literal string on Line " + lineNumber);
+                    if (i + 5 >= escaped.Length)
+                    {
+                        throw new ArgumentException($"There was a problem converting to literal string on Line {lineNumber}");
+                    }
```

Comment:
Yeah, I'd like to keep these as is.  I understand using exceptions when in non-test-related classes that just happens to be used by tests, as it's good to avoid the Xunit.Assert dependency.  But that's not the case here.  At best I think a change like this just breaks even, and at worst I think it makes the code less understandable.  Assert.* methods, after all, just do some work and generate an exception, so what we're really talking about here is adding code in order to change the exception type, and IMO it's not worth it, including the time it takes to develop and the time it takes to review such changes.